### PR TITLE
feat: enhance quote routing logic to avoid no route available errors for Base

### PR DESCRIPTION
**Pull request by Dprof-in-tech**
### Description

This pull request improves the bridge quote routing logic and its user interface feedback, ensuring that users only see "no route available" errors when a quote attempt has actually run and failed, rather than prematurely. It also refactors the quote-fetching logic to cleanly fall back to LI.FI when NEAR Intents cannot serve a route, and enhances test coverage for these routing cases.

**Bridge quote routing improvements:**

* Refactored the quote-fetching logic in `useBridgeQuote` to fall back to LI.FI when NEAR Intents lacks inventory for a given asset pair (e.g., USDT on Base, unsupported chains), ensuring more routes are available and reducing false negatives for users. [[1]](diffhunk://#diff-b064832579eb08d25e6ab81317021f046a3601d0285a92873d1d47db67b4e329L82-R127) [[2]](diffhunk://#diff-b064832579eb08d25e6ab81317021f046a3601d0285a92873d1d47db67b4e329L103-R148)
* Extracted LI.FI quote logic into a dedicated `fetchLifiQuote` function, centralizing slippage handling and native token address normalization.

**User interface and feedback enhancements:**

* Updated `BridgeForm` to distinguish between "no route available" due to an actual failed quote attempt and cases where the query never ran (e.g., wallet not connected), preventing premature error messages. This uses the new `isFetched` flag from `useBridgeQuote`. [[1]](diffhunk://#diff-d5ee3e53c4c66a5edce65affab6eac80a34b70368f880f4786f595c56d8e018aL157-R161) [[2]](diffhunk://#diff-d5ee3e53c4c66a5edce65affab6eac80a34b70368f880f4786f595c56d8e018aR174-R184) [[3]](diffhunk://#diff-d5ee3e53c4c66a5edce65affab6eac80a34b70368f880f4786f595c56d8e018aR330-R336) [[4]](diffhunk://#diff-b064832579eb08d25e6ab81317021f046a3601d0285a92873d1d47db67b4e329L138-R162) [[5]](diffhunk://#diff-b064832579eb08d25e6ab81317021f046a3601d0285a92873d1d47db67b4e329L155-R181)
* Improved engine selection logic in the UI to reflect the actual source of the quote after fallback, rather than just the predicted engine.

**Testing:**

* Added comprehensive tests for bridge quote routing, including fallback scenarios and edge cases, in `bridgeQuoteRouting.test.ts`.

- Introduced a new test suite for the useBridgeQuote hook, covering various scenarios including fallback mechanisms between NEAR and LI.FI.
- Updated BridgeForm component to improve quote handling and distinguish between fetched and non-fetched states.
- Refactored bridge.ts to streamline quote fetching logic, ensuring better handling of unsupported asset pairs and slippage configurations.
- Enhanced error handling and user feedback for scenarios where no routing options are available.

### References

closes KAN-739

### Testing

Screenshot 2026-08-10 at 7 55 50 am
Screenshot 2026-08-10 at 7 56 10 am

- [ ] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [ ] I have added documentation and tests for new/changed functionality in this PR
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used, if not `main`
- [ ] If this PR adds a database migration, it follows expand/contract: the new code works against the **pre-migration** schema, the currently deployed code keeps working against the **post-migration** schema, and destructive changes (drops, renames, tightened constraints) are deferred until the old application version is no longer serving — migrations are applied around the deploy, not strictly before or after it

By submitting a PR, I agree to Paycrest's [Contributor Code of Conduct](https://paycrest.notion.site/Contributor-Code-of-Conduct-1602482d45a2806bab75fd314b381f4c) and [Contribution Guide](https://paycrest.notion.site/Contribution-Guide-1602482d45a2809a8930e6ad565c906a).

## Summary by CodeRabbit

* **New Features**
  * Added automatic fallback to LI.FI when NEAR cannot support a selected asset.
  * Bridge engine selection now reflects the actual quote returned.
  * Added native-token handling and improved cNGN slippage support.

* **Bug Fixes**
  * Prevented premature “no route” warnings before wallet or authentication readiness.
  * Improved handling of unsuccessful quote attempts and empty results.

* **Tests**
  * Added coverage for quote routing, fallback behavior, native tokens, and loading states.

### CODE REVIEW

File: app/hooks/bridge.ts [hunk 1]
```diff
@@ -79,7 +119,12 @@ export function useBridgeQuote({
       const originAsset = resolveNearAssetId(from.token, from.network, tokenList);
       const destinationAsset = resolveNearAssetId(to.token, to.network, tokenList);
 
-      if (!originAsset || !destinationAsset) return null;
+      // NEAR Intents solvers only quote assets they hold inventory in, which leaves real
+      // gaps: no USDT on Base, and no Lisk/Celo/Tron at all. Those pairs are still routable
+      // through LI.FI's DEX aggregators, so fall back instead of reporting "no rail".
+      if (!originAsset || !destinationAsset) {
+        return fetchLifiQuote(from, to, rawAmount, evmAddress, slippageBps, auth);
+      }
```
[the remark this answers is not in the document]
Comment (Dprof-in-tech) [reply app/hooks/bridge.ts:127]: Both comments describe the same condition, and it can't occur. Skipped, no code changed.

Why

The finding asks me to await the NEAR quote and fall back when it's null. NearIntentsClient.getQuote never returns null:

// app/lib/bridge.ts:311-320
async getQuote(...): Promise<BridgeQuote> {          // non-nullable return type
  const { data } = await axios.post("/api/bridge/near-intents/quote", ...);
  return this.normalizeQuote(data, params, decimals); // always an object literal
}

Three things pin this down:
- The return type is Promise<BridgeQuote>, not Promise<BridgeQuote | null>.
- normalizeQuote (app/lib/bridge.ts:262-296) returns an object literal on every path; its only try/catch guards fee math and falls back to fee = "0".
- The axios.post uses default validateStatus, so a non-2xx throws. It does not resolve to null.

Contrast LifiClient.getQuote at line 386, which is Promise<BridgeQuote | null> and returns null on 404/4xx. I think the review conflated the two clients' contracts.

Adding ?? fetchLifiQuote(...) would be unreachable code, and the suggested test would have to mock getQuote into returning a value its own type forbids, so it would assert nothing about real behavior while looking like coverage.

**Review by sundayonah — APPROVED** [review]
looks good
